src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/GetCriteriaOperation.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.hmcts.ccd.domain.service.aggregated;
+
+import static uk.gov.hmcts.ccd.domain.model.search.CriteriaType.WORKBASKET;
+
+import uk.gov.hmcts.ccd.domain.model.definition.AccessControlList;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseField;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseType;
+import uk.gov.hmcts.ccd.domain.model.definition.CriteriaField;
+import uk.gov.hmcts.ccd.domain.model.search.CriteriaInput;
+import uk.gov.hmcts.ccd.domain.model.search.CriteriaType;
+import uk.gov.hmcts.ccd.domain.model.search.Field;
+import uk.gov.hmcts.ccd.domain.model.search.SearchInput;
+import uk.gov.hmcts.ccd.domain.model.search.WorkbasketInput;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+public interface GetCriteriaOperation {
+
+    <T> List<? extends CriteriaInput> execute(
+        final String caseTypeId,
+        Predicate<AccessControlList> access, CriteriaType criteriaType);
+
+    default CriteriaInput toCriteriaInput(final CriteriaField in, final CaseType caseType, CriteriaType criteriaType) {
+        CriteriaInput result;
+        if (criteriaType.equals(WORKBASKET)) {
```

Comment:
can't we just instantiate CriteriaInput? 
new CriteriaInput

Reply:
Issues with object conversion required for the FE. We may resolve it with another task to get rid of Workbasket/Search Input. See the next comment below. 👇